hack/externalnode/install-vm.ps1

```diff
@@ -0,0 +1,198 @@
+<#
+  .SYNOPSIS
+  Installs Antrea-Agent service.
+
+  .PARAMETER NameSpace
+  ExternalNode Namespace to be used.
+
+  .PARAMETER BinaryPath
+  Specifies the path of the antrea-agent binary to be used.
+
+  .PARAMETER ConfigPath
+  Specifies the path of the antrea-agent configuration file to be used.
+
+  .PARAMETER KubeConfigPath
+  Specifies the path of the kubeconfig to access K8s API Server.
+
+  .PARAMETER AntreaKubeConfigPath
+  Specifies the path of the kubeconfig to access Antrea API Server.
+
+  .PARAMETER InstallDir
+  The target installation directory. The default path is "C:\antrea-agent".
+#>
+Param(
+    [parameter(Mandatory = $true)] [string] $NameSpace,
+    [parameter(Mandatory = $true)] [string] $BinaryPath,
+    [parameter(Mandatory = $true)] [string] $ConfigPath,
+    [parameter(Mandatory = $true)] [string] $KubeConfigPath,
+    [parameter(Mandatory = $true)] [string] $AntreaKubeConfigPath,
+    [parameter(Mandatory = $false)] [string] $NodeName = $env:COMPUTERNAME,
+    [parameter(Mandatory = $false)] [string] $InstallDir = "C:\antrea-agent"
+)
+
+$ErrorActionPreference = "Stop"
+
+$WorkDir = [System.IO.Path]::GetDirectoryName($myInvocation.MyCommand.Definition)
+$InstallLog = "$WorkDir\install_vm.log"
+
+# Antrea paths
+$AntreaAgentPath = [io.path]::combine($InstallDir, "antrea-agent.exe")
+$AntreaAgentConfDir = [io.path]::combine($InstallDir, "conf")
+$AntreaAgentLogDir = [io.path]::combine($InstallDir, "logs")
+$AntreaAgentConfPath = [io.path]::combine($AntreaAgentConfDir, "antrea-agent.conf")
+$AntreaAgentLogFile = [io.path]::combine($AntreaAgentLogDir, "antrea-agent.log")
+
+# Constants
+$K8sKubeconfig = "antrea-agent.kubeconfig"
+$AntreaKubeconfig = "antrea-agent.antrea.kubeconfig"
+$OVSServices = "ovsdb-server", "ovs-vswitchd"
+$AntreaAgent = "antrea-agent"
+$Kubeconfig = "kubeconfig"
+$ExternalNodeNamespace = "externalNodeNamespace"
+
+function Log($Info) {
+    $time = $(get-date -Format g)
+    "$time $Info " | Tee-Object $InstallLog -Append | Write-Host
+}
+
+function ServiceExists($ServiceName) {
+    If (Get-Service $ServiceName -ErrorAction SilentlyContinue) {
+        return $true
+    }
+    return $false
+}
+
+function CheckSupportedVersions() {
+    $script:OS_VERSION = [System.Environment]::OSVersion.Version
```

Comment:
I think it is better to provide a list of supported versions, and pass the check if the current version is in the list. It is easier to maintain in future if we support more versions.  So is the Linux script

Reply:
Agree. For windows currently we only support one version. (i.e. we verified only on Windows Server 2019). 
While for Linux only ubuntu 18.04 and 20.04 is verified.

So I don't want to add a list as part of this PR. In the next release, when we support other version, then we can update the script to have a supported version list.